homeassistant/components/netwave/__init__.py

```diff
@@ -0,0 +1,211 @@
+                COMMAND_TURN_IO_OFF,
+                COMMAND_SET_PRESET,
+                COMMAND_RECALL_PRESET,
+                COMMAND_RESTART_CAMERA,
+                COMMAND_FACTORY_RESET_CAMERA,
+            ]
+        ),
+        vol.Optional(ATTR_PARAMETER, default=DEFAULT_PRESET): vol.Range(1, 15),
+        vol.Optional(ATTR_MOVE_DURATION, default=None): vol.Any(int, float, None),
+    }
+)
+
+SERVICE_PARAMETER_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_PARAMETER): vol.In(
+            [
+                PARAMETER_BRIGHTNESS,
+                PARAMETER_CONTRAST,
+                PARAMETER_MODE,
+                PARAMETER_ORIENTATION,
+                PARAMETER_RESOLUTION,
+            ]
+        ),
+        vol.Required(ATTR_VALUE): vol.Or(cv.string, cv.positive_int),
+    }
+)
+
+SERVICE_INFO_SCHEMA = vol.Schema({vol.Required(ATTR_ENTITY_ID): cv.entity_ids})
+
+
+async def _setup_services(hass: HomeAssistant):
+    """Initialize the services for NetWave."""
+    hass.data.setdefault(DOMAIN, {})
+
+    async def async_handle_command(call):
+        """Handle request to send a command."""
+        for entity in call.data[ATTR_ENTITY_ID]:
+            camera = hass.data[DOMAIN][entity]
+            await hass.async_add_executor_job(
+                camera.send_command, call.data[ATTR_COMMAND], call.data[ATTR_PARAMETER]
+            )
+            duration = (
+                camera.move_duration
+                if call.data[ATTR_MOVE_DURATION] is None
+                else call.data[ATTR_MOVE_DURATION]
+            )
+            if (
+                "move_" in call.data[ATTR_COMMAND]
+                and "center" not in call.data[ATTR_COMMAND]
+                and duration > 0
+            ):
+                await asyncio.sleep(duration)
+                await hass.async_add_executor_job(
+                    camera.send_command, COMMAND_STOP_MOVEMENT
+                )
+
+    async def async_handle_parameter(call):
+        """Handle request to update a camera parameter value."""
+        for entity in call.data[ATTR_ENTITY_ID]:
+            await hass.async_add_executor_job(
+                hass.data[DOMAIN][entity].send_parameter,
+                call.data[ATTR_PARAMETER],
+                call.data[ATTR_VALUE],
+            )
+
+    async def async_handle_info(call):
+        """Handle request to update info for a camera."""
+        for entity in call.data[ATTR_ENTITY_ID]:
+            await hass.async_add_executor_job(hass.data[DOMAIN][entity].get_info)
+
+    async def async_handle_refresh(call):
+        """Handle request to refresh all camera info."""
+        for _, camera in hass.data[DOMAIN].items():
+            await hass.async_add_executor_job(camera.get_info)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_COMMAND, async_handle_command, schema=SERVICE_COMMAND_SCHEMA
+    )
+    hass.services.async_register(
```

Comment:
These services can be greatly simplified when they are set up as entity services.

For more information see here: <https://developers.home-assistant.io/docs/dev_101_services#entity-services>